ovsdb/client.go

```diff
@@ -204,14 +212,23 @@ func (c *Client) doCallback(id int, res rpcResponse) {
 	c.cbMu.Lock()
 	defer c.cbMu.Unlock()
 
-	ch, ok := c.callbacks[id]
+	cb, ok := c.callbacks[id]
 	if !ok {
 		// Nobody is listening to this callback.
 		return
 	}
 
-	// Return result and remove this callback.
-	ch <- res
+	// Producer can safely close channel on return.
+	defer close(cb.Response)
+
+	// Wait for send or cancelation.
+	select {
+	case <-cb.Ctx.Done():
```

Comment:
Since we're using the `ctx.Done`, the consumer should check the receive channel with `case res, ok := <-ch`, as there might be a case that `close(ch)` wakes consumer than `ctx.Done()` on the consumer side.  It might be a super coner case but it's better to have that code explicit with the comment the reason why.

And looking for the test cases to simulate this senario.

Reply:
Yeah, and since we're already cleaning up the callback on the producer side (the `delete`, a few lines below this), I wonder if we can simplify the select on the consumer side, so it just returns `ctx.Err()` -- i.e. do we still need `cancelCallback` at this point?